src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -195,11 +195,13 @@ export function makeGenericAssayOption(
     // indentical to the entity_stable_id.
     const name = getGenericAssayMetaPropertyOrDefault(
         meta,
-        COMMON_GENERIC_ASSAY_PROPERTY.NAME
+        COMMON_GENERIC_ASSAY_PROPERTY.NAME,
+        meta.stableId
     );
```

Comment:
The reason why it was showing `NA`:
getGenericAssayMetaPropertyOrDefault will return `NA` when an genericAssayMetaProperty is missing.
```
export function getGenericAssayMetaPropertyOrDefault(
    meta: GenericAssayMeta,
    property: string,
    defaultValue: string = NOT_APPLICABLE_VALUE
): string {
    if (property in meta.genericEntityMetaProperties) {
        return (meta.genericEntityMetaProperties as {
            [property: string]: string;
        })[property];
    } else {
        return defaultValue;
    }
}
```